libc/spec/stdc.td

```diff
@@ -378,6 +378,7 @@ def StdC : StandardSpec<"stdc"> {
       [
           NamedType<"float_t">,
           NamedType<"double_t">,
+          Float16Type,
           NamedType<"float128">,
       ],
```

Comment:
Why weren't the type records from spec.td being reused here? E.g., why did we have `NamedType<"float128">` instead of `Float128Type`?